File: pubnub-gson/pubnub-gson-api/src/main/java/com/pubnub/api/java/endpoints/presence/SetState.java
package com.pubnub.api.java.endpoints.presence;

import com.pubnub.api.java.endpoints.Endpoint;
import com.pubnub.api.models.consumer.presence.PNSetStateResult;

public interface SetState extends Endpoint<PNSetStateResult> {
    SetState channels(java.util.List<String> channels);

    SetState channelGroups(java.util.List<String> channelGroups);

    SetState state(Object state);

    SetState uuid(String uuid);

    SetState withHeartbeat(boolean withHeartbeat);
}

File: pubnub-kotlin/pubnub-kotlin-api/src/jvmMain/kotlin/com/pubnub/api/endpoints/presence/SetState.kt
package com.pubnub.api.endpoints.presence

import com.pubnub.api.Endpoint
import com.pubnub.api.models.consumer.presence.PNSetStateResult

/**
 * @see [PubNub.setPresenceState]
 */
actual interface SetState : Endpoint<PNSetStateResult> {
    val channels: List<String>
    val channelGroups: List<String>
    val state: Any
    val uuid: String
}
